pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -326,7 +303,18 @@ func (i *inode) Open(ctx context.Context, rp *vfs.ResolvingPath, vfsd *vfs.Dentr
 		return nil, syserror.EOVERFLOW
 	}
 
-	fd := fileDescription{}
+	var fd *fileDescription
+	var fdImpl vfs.FileDescriptionImpl
+	if isDir {
```

Comment:
We'd better be careful with `isDir` here. If a malicious FUSE daemon send mismatching node attr, it will potentially crash the sentry